include/handler.h

```diff
@@ -476,6 +480,19 @@ class handler {
 		m_split_constraint = detail::range_cast<3>(constraint);
 	}
 
+	template <typename Hint>
+	void experimental_hint(Hint&& hint) {
+		static_assert(std::is_base_of_v<detail::hint_base, std::decay_t<Hint>>, "Incompatible hint");
```

Comment:
```suggestion
		static_assert(std::is_base_of_v<detail::hint_base, Hint>, "Hint must extend hint_base");
```
`task::get_hint()` has a better error message for this case.